### PR TITLE
Fixes #1363

### LINKED ISSUE #1363: Fix the disable users by tenancy migration
Doesn't run due to a dependency error

### PR DESCRIPTION
**Pull request by brecke**
Added check to avoid dependency error
Added a log error with the uncaughtException stack trace
Minor fixes to the migration

Test it by running `node etc/migration/disable_users_from_tenancy/1.disable-users-by-tenancy.js -t someTenant | bunyan`

### CODE REVIEW

**Comment by brecke** [comment]
Refactored it a bit

File: node_modules/oae-principals/lib/model.user.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ var TenantsAPI = require('oae-tenants');
  * @param  {Boolean}    [opts.isTenantAdmin]            Whether or not the user is a tenant admin
  */
 module.exports.User = function(tenantAlias, id, displayName, email, opts) {
+    var TenantsAPI = TenantsAPI || require('oae-tenants');
```
Comment (sallakarppinen) [inline node_modules/oae-principals/lib/model.user.js]: Can we take this out if we are generating a context in the migration script?

File: etc/migration/disable_users_from_tenancy/lib/disable-users-by-tenancy.js [hunk 2]
```diff
@@ -29,6 +37,9 @@ var PrincipalsAPI = require('oae-principals');
  * @param  {Function}   callback        Standard callback function
  */
 var doMigration = function (ctx, tenantAlias, disabled, callback) {
+
+    ctx = ctx || _createNewContext();
```
Comment (sallakarppinen) [inline etc/migration/disable_users_from_tenancy/lib/disable-users-by-tenancy.js]: This does seem a bit unsafe but should be OK for a migration.

File: node_modules/oae-principals/lib/model.user.js (hunk 1 above)
Comment (brecke) [inline node_modules/oae-principals/lib/model.user.js]: oops. I forgot to remove this

File: etc/migration/disable_users_from_tenancy/lib/disable-users-by-tenancy.js (hunk 2 above)
Comment (brecke) [inline etc/migration/disable_users_from_tenancy/lib/disable-users-by-tenancy.js]: why unsafe?

Comment (sallakarppinen) [inline etc/migration/disable_users_from_tenancy/lib/disable-users-by-tenancy.js]: Well, `_createNewContext` generates a new global admin context which would be not-good anywhere except in a migration script, OK here though.

File: etc/migration/disable_users_from_tenancy/1.disable-users-by-tenancy.js [hunk 3]
```diff
@@ -83,19 +81,14 @@ OAE.init(config, function (err) {
...
-  var globalAdmin = new User(globalTenant.alias, util.format('u:%s:admin', globalTenant.alias), 'Global Administrator', null, {
-    'visibility': AuthzConstants.visibility.PRIVATE,
-    'isGlobalAdmin': true
-  });
 
-  DisableUsersMigration.doMigration(new Context(globalTenant, globalAdmin), tenantAlias, true, function(err, users) {
+  DisableUsersMigration.doMigration(null, tenantAlias, true, function(err, users) {
     if (err) {
-      log().warn('Migration not completed successfully.');
+      log().warn("Migration not completed successfully.");
```
Comment (sallakarppinen) [inline etc/migration/disable_users_from_tenancy/1.disable-users-by-tenancy.js]: Should use single quotes.

File: etc/migration/disable_users_from_tenancy/1.disable-users-by-tenancy.js [hunk 4]
```diff
@@ -83,19 +81,14 @@ OAE.init(config, function (err) {
...
       process.exit(err.code);
     }
 
-    log().info('Finished migration for ' + users.length + ' users.');
+    log().info("Finished migration for " + users.length + " users.");
```
Comment (sallakarppinen) [inline etc/migration/disable_users_from_tenancy/1.disable-users-by-tenancy.js]: Here as well, single quotes for all.

File: node_modules/oae-util/lib/oae.js [hunk 5]
```diff
@@ -61,6 +61,7 @@ module.exports.init = function(config, callback) {
 
     // Set up application-level error handler
     process.on('uncaughtException', function(err) {
+        log().error(err.stack);
```
Comment (sallakarppinen) [inline node_modules/oae-util/lib/oae.js]: These logging statements could be swapped around so that the stack trace shows up below the message.

**Pull request merged**